src/js/stores/dcos-version.ts

```diff
@@ -0,0 +1,55 @@
+import { from } from "rxjs";
+import { fromFetch } from "rxjs/fetch";
+import { delay, switchMap, shareReplay, retryWhen, tap } from "rxjs/operators";
+import Config from "#SRC/js/config/Config";
+
+import container from "#SRC/js/container";
+import {
+  ToastNotification,
+  ToastAppearance,
+} from "@extension-kid/toast-notifications";
+import {
+  NotificationService,
+  NotificationServiceType,
+} from "@extension-kid/notification-service";
+
+const notificationService = container.get<NotificationService>(
+  NotificationServiceType
+);
+
+type ClusterData = {
+  bootstrapId: string;
+  imageCommit: string;
+  variant: string;
+  version: string;
+};
+
+export default fromFetch(
+  `${Config.rootUrl}/dcos-metadata/dcos-version.json`
+).pipe(
+  switchMap((response) =>
+    from(
+      response.json().then((json) => ({
+        version: json.version,
+        imageCommit: json["dcos-image-commit"],
+        bootstrapId: json["bootstrap-id"],
+        variant: json["dcos-variant"],
+      }))
+    )
+  ),
+  retryWhen((errors) =>
+    errors.pipe(
+      tap((e) => {
+        notificationService.push(
+          new ToastNotification("Error while fetching data", {
+            appearance: ToastAppearance.Danger,
+            description: `An error occurred while fetching ${Config.rootUrl}/dcos-metadata/dcos-version.json. Here's the internal error: ${e}`,
+          })
+        );
+      }),
+      // retry every 10 seconds
+      delay(10000)
+    )
+  ),
+  shareReplay<ClusterData>(1)
+);
```

Comment:
@orlandohohmeier review requested mainly for this part. 

we're trying forever with a delay of 10 seconds, show a toast in case something failed and `shareReplay` the result amongst subscribers.

Now that i see the stuff above i realize i forgot to i18n'ize. Will do that soon. 😊

Reply:
🤔 What do you expect users do with the information?

Reply:
File better COPS issues...

Reply:
would merging be ok with you or do we want to do something different in the case of an error. till now every page would have remained in a loading state  in case that endpoint was not available. so showing an error that might be confusing or too detailed seems still better than not doing anything at all, right?

Reply:
As discussed offline, let's adjust this to apply capped exponential back-off plus a a message after about 5s to not block interaction any longer then needed while waiting for the response.